modules/glean/runtime/glean-plugin.client.ts

```diff
@@ -33,24 +33,41 @@ export default defineNuxtPlugin((nuxtApp) => {
       }
     }
 
+    function recordEngagement(element: Element) {
+      if (!element)
+        return
+
+      if (!element.hasAttribute('data-glean'))
+        return
+
+      const data = element.getAttribute('data-glean') || ''
+      const value = element.getAttribute('data-glean-value') || ''
+      engagement.record({ ui_identifier: data, engagement_value: value, ...engagementDetails[data] })
+    }
+
     function handleButtonClick(ev: MouseEvent) {
       const eventTarget = ev?.target as Element
       const closestButton = eventTarget.closest('button')
 
+      if (!closestButton)
```

Comment:
Added an early return here that makes more sense than guarding just `recordEngagement` as so - 

```
if (closestButton)
  recordEngagement(eventTarget)
```

I'm a big fan of avoiding indentation levels